src/sentry/api/endpoints/external_team.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from rest_framework import serializers, status
+from rest_framework.response import Response
+
+from sentry.api.bases.team import TeamEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import ExternalTeam, EXTERNAL_PROVIDERS
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalTeamSerializer(serializers.ModelSerializer):
+    provider = serializers.ChoiceField(choices=list(EXTERNAL_PROVIDERS.values()))
+
+    class Meta:
+        model = ExternalTeam
+        fields = ["team_id", "external_id", "provider"]
+
+    def validate_provider(self, provider):
+        if provider not in EXTERNAL_PROVIDERS.values():
+            raise serializers.ValidationError(
+                f'The provider "{provider}" is not supported. We currently accept Github and Gitlab team identities.'
```

Comment:
@NisanthanNanthakumar Nit: we will have to update this error message once we add new platforms but maybe that's OK (we just need to remember to do this).